File: src/main/java/org/itmo/isLab1/people/PersonController.java
package org.itmo.isLab1.people;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;
import org.itmo.isLab1.common.framework.CrudController;
import org.itmo.isLab1.people.dto.*;

@RestController
@RequestMapping(value = "/api/people", produces = MediaType.APPLICATION_JSON_VALUE)
public class PersonController
    extends CrudController<
        Person,
        PersonDto,
        PersonCreateDto,
        PersonUpdateDto,
        PersonService> {

    public PersonController(
        PersonService service
    ) {
        super(service);
    }
}

File: frontend/src/controllers/PersonController.ts
import { PersonDto } from '@/interfaces/dto/people/PersonDto';
import { PersonCreateDto } from '@/interfaces/dto/people/PersonCreateDto';
import { PersonUpdateDto } from '@/interfaces/dto/people/PersonUpdateDto';
import CrudController from '@/interfaces/crud/CrudController';

export default new CrudController<PersonDto, PersonCreateDto, PersonUpdateDto>("people")
